apply() does not work on iterable

Per #473: `apply()` should work with iterable.
E.g.: the following code does not compile:

```typescript
const minValue = execPipe(
  [2, 7, 14],
  map(x => x * x),
  apply(Math.min)
);
```

> error TS2769: No overload matches this call.
  Overload 1 of 8, '(initial: number[], fn0: (value: number[]) => any[] | null | undefined, fn1: (value: any[] | null | undefined) => number): number', gave the following error.       
    Argument of type '(source: Wrappable<number>) => IterableIterator<number>' is not assignable to parameter of type '(value: number[]) => any[] | null | undefined'.

--- a/src/impls/apply/__tests__/apply.test.ts
+++ b/src/impls/apply/__tests__/apply.test.ts
@@ -1,17 +1,19 @@
 /* eslint-disable jest/expect-expect */
 import { apply } from 'iter-tools-es';
 
+import { wrap } from '../../../test/helpers.js';
+
 describe('apply', () => {
   it('passes the function the iterable of arguments provided to it', () => {
     const testFn = jest.fn();
-    apply(testFn, [2, 3]);
+    apply(testFn, wrap([2, 3]));
     expect(testFn).toHaveBeenCalledTimes(1);
     expect(testFn).toHaveBeenLastCalledWith(2, 3);
   });
 
   it('can be curried', () => {
     const testFn = jest.fn();
-    apply(testFn)([2, 3]);
+    apply(testFn)(wrap([2, 3]));
     expect(testFn).toHaveBeenCalledTimes(1);
     expect(testFn).toHaveBeenLastCalledWith(2, 3);
   });

--- a/src/impls/apply/apply.d.ts
+++ b/src/impls/apply/apply.d.ts
@@ -1,7 +1,6 @@
-declare function apply<Return, A>(
-  fn: (...args: Array<A>) => Return,
-  args: Iterable<A> | null | undefined,
-): Return;
+import { Wrappable } from 'src/types';
+
+declare function apply<Return, A>(fn: (...args: Array<A>) => Return, args: Wrappable<A>): Return;
 
 declare function apply<Args extends Array<any>, Return>(
   fn: (...args: Args) => Return,
@@ -10,6 +9,6 @@ declare function apply<Args extends Array<any>, Return>(
 
 declare function apply<Return>(
   fn: (...args: Array<any>) => Return,
-): (args: Array<any> | null | undefined) => Return;
+): (args: Wrappable<any>) => Return;
 
 export { apply };

